File: java/binary_search.java
import java.util.Scanner;

public class binary_search {

	public static void main(String[] args) {

		int[] array = new int[10];
		int i, n, inicio, meio, fim;
		boolean achou;

		for(i=0;i<10;i++) {

			array[i] = i+1;
		}

		Scanner s = new Scanner(System.in);
		System.out.print("Type the number to look for: ");
		n = s.nextInt();

		achou = false;
		inicio = 0;
		fim = 9;
		meio = (inicio+fim)/2;

		while(!achou && inicio <= fim) {

			if(array[meio] == n)
				achou = true;
			else {
				if(n < array[meio])
					fim = meio - 1;
				else
					inicio = meio + 1;
				meio = (inicio+fim)/2;
			}
		}

		if(achou)
			System.out.println("Number found at position " + (meio + 1));
		else
			System.out.println("Number not found");
	}
}

File: cpp/binary_search.cpp
#include <iostream>
#include <stdbool.h>

#define ARRAY_LENGTH 1000000

int main() {

	int array[ARRAY_LENGTH], i, n, inicio, meio, fim;
	bool achou;

	//load the vector numbers:
	for(i=0; i<ARRAY_LENGTH; i++)
	{
		array[i] = i+1;
	}

	std::cout << "Type the number to look for: ";
	std::cin >> n;

	achou = false;
	inicio = 0;
	fim = ARRAY_LENGTH - 1;
	meio = (inicio+fim)/2;

	while(inicio <= fim && !achou) {

		if(array[meio] == n)
			achou = true;
		else {
			if(n < array[meio]) fim = meio - 1;
			else inicio = meio + 1;
			meio = (inicio+fim)/2;
		}
	}

	if(achou) std::cout << "Number found at position " << meio + 1 << std::endl;
	else std::cout << "Number not found" << std::endl;

	return 0;
}
